lib/Client.js

```diff
@@ -2172,6 +2172,15 @@ class Client extends EventEmitter {
         return this.requestHandler.request("PATCH", Endpoints.GUILD_ROLE(guildID, roleID), true, options).then((role) => new Role(role, this.guilds.get(guildID)));
     }
 
+    /**
+     * Update the role connection metadata for the application
+     * @arg {Array<Object>} data An array of role connection metadata objects. See [the official Discord API documentation entry](https://discord.com/developers/docs/resources/application-role-connection-metadata#application-role-connection-metadata-object) for object structure
+     * @returns {Promise<Array<Object>>} An array of role connection metadata objects
+     */
+    editRoleConnectionMetadataRecords(data) {
+        return this.requestHandler.request("PUT", Endpoints.ROLE_CONNECTION_METADATA_RECORDS(this.application.id), true, data);
```

Comment:
May be worth regex validating `key`. We do the same thing for application command names

Reply:
Sure. How's that?